sdk/storage/MigrationGuide.md

```diff
@@ -0,0 +1,477 @@
+
+## Table of contents
+
+- [Migration benefits](#migration-benefits)
+- [General changes](#general-changes)
+  - [Package and namespaces](#package-and-namespaces)
+  - [Authentication](#authentication)
+  - [Client structure](#client-structure)
+- [Migration samples](#migration-samples)
+  - [Creating a Container](#creating-a-container)
+  - [Uploading Blobs to a Container](#uploading-blobs-to-a-container)
+  - [Downloading Blobs from a Container](#downloading-blobs-from-a-container)
+  - [Listing Blobs in a Container](#listing-blobs-in-a-container)
+  - [Managing Blob Metadata](#managing-blob-metadata)
+  - [Content Hashes](#content-hashes)
+  - [Resiliency](#resiliency)
+- [Additional information](#additional-information)
+
+## Migration benefits
+
+To understand why we created our version 12 client libraries, you may refer to the Tech Community blog post, [Announcing the Azure Storage v12 Client Libraries](https://techcommunity.microsoft.com/t5/azure-storage/announcing-the-azure-storage-v12-client-libraries/ba-p/1482394) or refer to our video [Introducing the New Azure SDKs](https://aka.ms/azsdk/intro).
+
+Included are the following:
+- Thread-safe synchronous APIs
+- Improved performance
+- Consistent and idiomatic code organization, naming, and API structure, aligned with a set of common guidelines
+- The learning curve associated with the libraries was reduced
+
+Note: The blog post linked above announces deprecation for previous versions of the library.
+
+## General changes
+
+### Package and namespaces
+
+Package names and the namespaces root for version 12 Azure client libraries follow the pattern `Azure::[Area]::[Service]` where the legacy libraries followed the pattern `azure::[area]::[service]`.
+
+In this case, to install the legacy v7.5 package with vcpkg:
+```bash
+vcpkg install azure-storage-cpp
+```
+
+It is now the following for v12:
+```bash
+vcpkg install azure-storage-blobs-cpp
+```
+
+### Authentication
+
+#### Azure Active Directory
+
+v7.5
+
+The legacy Storage SDK contained a `bearer_token_credential` class that could be used to populate a `storage_credentials` instance. Constructors took a string token for HTTP authorization headers and an optional refresh mechanism for the library to invoke when the token expired.
+
+v12
+
+A `TokenCredential` abstract class (different API surface than v7.5) exists in the [Azure Core](https://github.com/Azure/azure-sdk-for-cpp/tree/master/sdk/core/azure-core) package that all libraries of the new Azure SDK family depend on, and can be used to construct Storage clients. Implementations of this class can be found separately in the [Azure Identity](https://github.com/Azure/azure-sdk-for-cpp/tree/master/sdk/identity/azure-identity) package.
+
+```C++
+BlobServiceClient serviceClient(serviceUrl, std::make_shared<Azure::Identity::ClientSecretCredential>(tenantId, clientId, clientSecret));
```

Comment:
How `serviceUrl` is got and all other URLs are constructed is a very important thing that should be mentioned. I don't think track 1 users are familiar with this field.

Reply:
Familiarity with the legacy client library is assumed.

If legacy client customers are not familiar with URL, how did they construct a client authed with bear token?

Reply:
As long as they are using track 1 SDK to get the URL, it does not work for track 2 right?